### ISSUE TITLE
Not working in Jupyter (IPython 4)

### ISSUE THREAD

**Issue by lrq3000** [flying-sheep/smart-progress#1]
Hi there,

Interesting work you did here, but it doesn't work on my setup, and I don't really know why. The only thing I know is that the culprit seems to be coming from `click` which also produces the same behavior when I try to directly call it: hidden output inside IPython, but it works in the console.

Also, there are a few bugs you might want to fix:
- In `__init__()`, remove the `*,` just before `label=None`.
- replace `super().__init__(...)` by `super(ProgressBar, self).__init__(...)`.

Good continuation for your project!

**Comment by flying-sheep** [comment]
could you try to reproduce this with python 3?

**Comment by lrq3000** [comment]
No, sorry I don't have IPython on Python 3, and I can't afford to install it (I have a SSD! :( ).

**Comment by lrq3000** [comment]
But indeed you guessed well, I use Python v2.7.10 on Windows 7 x64.

**Comment by flying-sheep** [comment]
i didn’t have to guess, your “bugs” are python 3 features :smile: 

maybe i can use [python-future](http://python-future.org/) to make the superior `super()` available for python2 users because i’m sure as hell not uglifying my code for a 6 years obsolete language version.

---

@ bug: this is probably caused by my approach of simply [`try`ing to create the IPython progressbar](https://github.com/flying-sheep/smart-progress/blob/b5a865bcca0a36e375cd36ea148c2cf2ed0c8c5e/smart_progress.py#L63-L66).

can you try to simply `from smart_progress import IPyBackend` and then instantiate a `IPyBackend(...)` with the same arguments you used? (you need to match them!)

**Comment by lrq3000** [comment]
I'll try that and I'll report the results here :)
